### ISSUE TITLE
Vis 1.5.1 cannot be installed if no vis exists at system

### ISSUE THREAD

**Issue by mcm1957** [ioBroker/ioBroker.vis#805]
**Describe the bug**  
Installing vis 1.5.1 as UPDATE works as expected.
INstalling vis 1.5.1 without a previous vis installation fails

**To Reproduce**  
1. Take a system with installed vis 1.5.1 and repo set to latest
2. Deinstall vis by using 'iob del vis'
3. Try to install vis 1.5.1 by using 'iob i vis'
    This installation fails (see below)
4. Try to install vis 1.4.16 by using 'iob i vis@1,4,16'
    This installation succeeds
5. Try to update vis to 1.5.1 by using 'iob update vis'
    This installation succeeds

**Expected behavior**  
Direct installation of 1.5.1 should woirk

**Screenshots & Logfiles**  

Failing installation of 1.5.1 at a system without vis reports:
```
pi@pi4:~ $ iob i vis
NPM version: 9.8.1
Installing iobroker.vis@1.5.1... (System call)
npm ERR! code 6
npm ERR! path /opt/iobroker/node_modules/iobroker.vis
npm ERR! command failed
npm ERR! command sh -c node main.js --install

npm ERR! A complete log of this run can be found in: /home/iobroker/.npm/_logs/2023-11-25T22_29_36_720Z-debug-0.log
host.pi4 Cannot install iobroker.vis@1.5.1: 6
pi@pi4:~ $
```

```
pi@pi4:~ $ cat  /home/iobroker/.npm/_logs/2023-11-25T22_23_00_740Z-debug-0.log
0 verbose cli /usr/bin/node /usr/bin/npm
1 info using npm@9.8.1
2 info using node@v18.18.2
3 timing npm:load:whichnode Completed in 5ms
4 timing config:load:defaults Completed in 6ms
5 timing config:load:file:/usr/lib/node_modules/npm/npmrc Completed in 14ms
6 timing config:load:builtin Completed in 14ms
7 timing config:load:cli Completed in 9ms
8 timing config:load:env Completed in 1ms
9 timing config:load:file:/opt/iobroker/.npmrc Completed in 6ms
10 timing config:load:project Completed in 8ms
11 timing config:load:file:/home/iobroker/.npmrc Completed in 1ms
12 timing config:load:user Completed in 1ms
13 timing config:load:file:/usr/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 1ms
15 timing config:load:setEnvs Completed in 3ms
16 timing config:load Completed in 44ms
17 timing npm:load:configload Completed in 46ms
18 timing config:load:flatten Completed in 9ms
19 timing npm:load:mkdirpcache Completed in 1ms
20 timing npm:load:mkdirplogs Completed in 1ms
21 verbose title npm install iobroker.vis@1.5.1
22 verbose argv "install" "--save-exact" "--loglevel" "error" "iobroker.vis@1.5.1"
23 timing npm:load:setTitle Completed in 3ms
24 timing npm:load:display Completed in 2ms
25 verbose logfile logs-max:10 dir:/home/iobroker/.npm/_logs/2023-11-25T22_23_00_740Z-
26 verbose logfile /home/iobroker/.npm/_logs/2023-11-25T22_23_00_740Z-debug-0.log
27 timing npm:load:logFile Completed in 26ms
28 timing npm:load:timers Completed in 1ms
29 timing npm:load:configScope Completed in 0ms
30 timing npm:load Completed in 140ms
31 timing arborist:ctor Completed in 2ms
32 silly logfile start cleaning logs, removing 2 files
33 silly logfile done cleaning log files
34 timing idealTree:init Completed in 3482ms
35 timing idealTree:userRequests Completed in 6ms
36 silly idealTree buildDeps
37 silly fetch manifest iobroker.vis@1.5.1
38 verbose shrinkwrap failed to load node_modules/.package-lock.json out of date, updated: node_modules
39 http fetch GET 200 https://registry.npmjs.org/iobroker.vis 2675ms (cache revalidated)
40 silly placeDep ROOT iobroker.vis@1.5.1 OK for: iobroker.inst@3.0.0 want: 1.5.1
41 silly fetch manifest @iobroker/adapter-core@^3.0.4
42 http fetch GET 200 https://registry.npmjs.org/@iobroker%2fadapter-core 1750ms (cache revalidated)
43 timing idealTree:#root Completed in 4480ms
44 silly fetch manifest @iobroker/types@^5.0.11
45 http fetch GET 200 https://registry.npmjs.org/@iobroker%2ftypes 1042ms (cache revalidated)
46 silly placeDep node_modules/iobroker.vis @iobroker/adapter-core@3.0.4 OK for: iobroker.vis@1.5.1 want: ^3.0.4
47 timing idealTree:node_modules/iobroker.vis Completed in 1062ms
48 timing idealTree:node_modules/iobroker.vis/node_modules/@iobroker/adapter-core Completed in 0ms
49 timing idealTree:buildDeps Completed in 5544ms
50 timing idealTree:fixDepFlags Completed in 46ms
51 timing idealTree Completed in 9124ms
52 timing reify:loadTrees Completed in 9127ms
53 timing reify:diffTrees Completed in 86ms
54 silly reify moves {}
55 timing reify:retireShallow Completed in 3ms
56 timing reify:createSparse Completed in 11ms
57 timing reify:loadBundles Completed in 0ms
58 verbose reify failed optional dependency /opt/iobroker/node_modules/osx-temperature-sensor
59 silly reify mark deleted [ '/opt/iobroker/node_modules/osx-temperature-sensor' ]
60 verbose reify failed optional dependency /opt/iobroker/node_modules/node-aead-crypto
61 silly reify mark deleted [ '/opt/iobroker/node_modules/node-aead-crypto' ]
62 verbose reify failed optional dependency /opt/iobroker/node_modules/node-aead-crypto/node_modules/nan
63 silly reify mark deleted [ '/opt/iobroker/node_modules/node-aead-crypto/node_modules/nan' ]
64 verbose reify failed optional dependency /opt/iobroker/node_modules/prebuild-install
65 silly reify mark deleted [
65 silly reify   '/opt/iobroker/node_modules/prebuild-install',
65 silly reify   '/opt/iobroker/node_modules/.bin/prebuild-install'
65 silly reify ]
66 verbose reify failed optional dependency /opt/iobroker/node_modules/detect-libc
67 silly reify mark deleted [
67 silly reify   '/opt/iobroker/node_modules/detect-libc',
67 silly reify   '/opt/iobroker/node_modules/.bin/detect-libc'
67 silly reify ]
68 verbose reify failed optional dependency /opt/iobroker/node_modules/expand-template
69 silly reify mark deleted [ '/opt/iobroker/node_modules/expand-template' ]
70 verbose reify failed optional dependency /opt/iobroker/node_modules/github-from-package
71 silly reify mark deleted [ '/opt/iobroker/node_modules/github-from-package' ]
72 verbose reify failed optional dependency /opt/iobroker/node_modules/napi-build-utils
73 silly reify mark deleted [ '/opt/iobroker/node_modules/napi-build-utils' ]
74 verbose reify failed optional dependency /opt/iobroker/node_modules/node-abi
75 silly reify mark deleted [ '/opt/iobroker/node_modules/node-abi' ]
76 verbose reify failed optional dependency /opt/iobroker/node_modules/npmlog
77 silly reify mark deleted [ '/opt/iobroker/node_modules/npmlog' ]
78 verbose reify failed optional dependency /opt/iobroker/node_modules/rc
79 silly reify mark deleted [
79 silly reify   '/opt/iobroker/node_modules/rc',
79 silly reify   '/opt/iobroker/node_modules/.bin/rc'
79 silly reify ]
80 verbose reify failed optional dependency /opt/iobroker/node_modules/simple-get
81 silly reify mark deleted [ '/opt/iobroker/node_modules/simple-get' ]
82 verbose reify failed optional dependency /opt/iobroker/node_modules/tar-fs
83 silly reify mark deleted [ '/opt/iobroker/node_modules/tar-fs' ]
84 verbose reify failed optional dependency /opt/iobroker/node_modules/node-abi/node_modules/semver
85 silly reify mark deleted [
85 silly reify   '/opt/iobroker/node_modules/node-abi/node_modules/semver',
85 silly reify   '/opt/iobroker/node_modules/node-abi/node_modules/.bin/semver'
85 silly reify ]
86 verbose reify failed optional dependency /opt/iobroker/node_modules/are-we-there-yet
87 silly reify mark deleted [ '/opt/iobroker/node_modules/are-we-there-yet' ]
88 verbose reify failed optional dependency /opt/iobroker/node_modules/gauge
89 silly reify mark deleted [ '/opt/iobroker/node_modules/gauge' ]
90 verbose reify failed optional dependency /opt/iobroker/node_modules/deep-extend
91 silly reify mark deleted [ '/opt/iobroker/node_modules/deep-extend' ]
92 verbose reify failed optional dependency /opt/iobroker/node_modules/strip-json-comments
93 silly reify mark deleted [ '/opt/iobroker/node_modules/strip-json-comments' ]
94 verbose reify failed optional dependency /opt/iobroker/node_modules/decompress-response
95 silly reify mark deleted [ '/opt/iobroker/node_modules/decompress-response' ]
96 verbose reify failed optional dependency /opt/iobroker/node_modules/simple-concat
97 silly reify mark deleted [ '/opt/iobroker/node_modules/simple-concat' ]
98 verbose reify failed optional dependency /opt/iobroker/node_modules/chownr
99 silly reify mark deleted [ '/opt/iobroker/node_modules/chownr' ]
100 verbose reify failed optional dependency /opt/iobroker/node_modules/tar-stream
101 silly reify mark deleted [ '/opt/iobroker/node_modules/tar-stream' ]
102 verbose reify failed optional dependency /opt/iobroker/node_modules/are-we-there-yet/node_modules/readable-stream
103 silly reify mark deleted [
103 silly reify   '/opt/iobroker/node_modules/are-we-there-yet/node_modules/readable-stream'
103 silly reify ]
104 verbose reify failed optional dependency /opt/iobroker/node_modules/mimic-response
105 silly reify mark deleted [ '/opt/iobroker/node_modules/mimic-response' ]
106 verbose reify failed optional dependency /opt/iobroker/node_modules/fs-constants
107 silly reify mark deleted [ '/opt/iobroker/node_modules/fs-constants' ]
108 verbose reify failed optional dependency /opt/iobroker/node_modules/tar-stream/node_modules/readable-stream
109 silly reify mark deleted [
109 silly reify   '/opt/iobroker/node_modules/tar-stream/node_modules/readable-stream'
109 silly reify ]
110 verbose reify failed optional dependency /opt/iobroker/node_modules/are-we-there-yet/node_modules/isarray
111 silly reify mark deleted [ '/opt/iobroker/node_modules/are-we-there-yet/node_modules/isarray' ]
112 verbose reify failed optional dependency /opt/iobroker/node_modules/are-we-there-yet/node_modules/string_decoder
113 silly reify mark deleted [
113 silly reify   '/opt/iobroker/node_modules/are-we-there-yet/node_modules/string_decoder'
113 silly reify ]
114 verbose reify failed optional dependency /opt/iobroker/node_modules/tar-stream/node_modules/string_decoder
115 silly reify mark deleted [ '/opt/iobroker/node_modules/tar-stream/node_modules/string_decoder' ]
116 verbose reify failed optional dependency /opt/iobroker/node_modules/tar-stream/node_modules/safe-buffer
117 silly reify mark deleted [ '/opt/iobroker/node_modules/tar-stream/node_modules/safe-buffer' ]
118 verbose reify failed optional dependency /opt/iobroker/node_modules/fsevents
119 silly reify mark deleted [ '/opt/iobroker/node_modules/fsevents' ]
120 timing reifyNode:node_modules/osx-temperature-sensor Completed in 46ms
121 timing reifyNode:node_modules/node-aead-crypto Completed in 47ms
122 timing reifyNode:node_modules/fsevents Completed in 46ms
123 timing reifyNode:node_modules/strip-json-comments Completed in 532ms
124 timing reifyNode:node_modules/chownr Completed in 535ms
125 timing reifyNode:node_modules/simple-get Completed in 544ms
126 timing reifyNode:node_modules/mimic-response Completed in 544ms
127 timing reifyNode:node_modules/fs-constants Completed in 546ms
128 timing reifyNode:node_modules/decompress-response Completed in 546ms
129 timing reifyNode:node_modules/expand-template Completed in 553ms
130 timing reifyNode:node_modules/tar-stream/node_modules/string_decoder Completed in 559ms
131 timing reifyNode:node_modules/simple-concat Completed in 560ms
132 timing reifyNode:node_modules/npmlog Completed in 562ms
133 timing reifyNode:node_modules/napi-build-utils Completed in 562ms
134 timing reifyNode:node_modules/deep-extend Completed in 563ms
135 timing reifyNode:node_modules/are-we-there-yet/node_modules/string_decoder Completed in 566ms
136 timing reifyNode:node_modules/tar-stream/node_modules/safe-buffer Completed in 575ms
137 timing reifyNode:node_modules/are-we-there-yet/node_modules/isarray Completed in 571ms
138 timing reifyNode:node_modules/detect-libc Completed in 581ms
139 timing reifyNode:node_modules/are-we-there-yet Completed in 583ms
140 timing reifyNode:node_modules/tar-stream Completed in 588ms
141 timing reifyNode:node_modules/node-abi/node_modules/semver Completed in 590ms
142 timing reifyNode:node_modules/iobroker.vis/node_modules/@iobroker/adapter-core Completed in 604ms
143 timing reifyNode:node_modules/github-from-package Completed in 603ms
144 timing reifyNode:node_modules/rc Completed in 607ms
145 timing reifyNode:node_modules/prebuild-install Completed in 610ms
146 timing reifyNode:node_modules/node-abi Completed in 610ms
147 timing reifyNode:node_modules/gauge Completed in 629ms
148 timing reifyNode:node_modules/are-we-there-yet/node_modules/readable-stream Completed in 649ms
149 timing reifyNode:node_modules/tar-stream/node_modules/readable-stream Completed in 657ms
150 timing reifyNode:node_modules/tar-fs Completed in 661ms
151 timing reifyNode:node_modules/node-aead-crypto/node_modules/nan Completed in 681ms
152 timing reifyNode:node_modules/iobroker.vis Completed in 1005ms
153 timing reify:unpack Completed in 1006ms
154 timing reify:unretire Completed in 0ms
155 timing build:queue Completed in 6ms
156 timing build:link Completed in 1ms
157 info run iobroker.vis@1.5.1 install node_modules/iobroker.vis node main.js --install
158 info run iobroker.vis@1.5.1 install { code: 6, signal: null }
159 timing reify:rollback:createSparse Completed in 107ms
160 timing reify:rollback:retireShallow Completed in 0ms
161 timing command:install Completed in 14804ms
162 verbose stack Error: command failed
162 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
162 verbose stack     at ChildProcess.emit (node:events:517:28)
162 verbose stack     at maybeClose (node:internal/child_process:1098:16)
162 verbose stack     at ChildProcess._handle.onexit (node:internal/child_process:303:5)
163 verbose pkgid iobroker.vis@1.5.1
164 verbose cwd /opt/iobroker
165 verbose Linux 6.1.21-v8+
166 verbose node v18.18.2
167 verbose npm  v9.8.1
168 error code 6
169 error path /opt/iobroker/node_modules/iobroker.vis
170 error command failed
171 error command sh -c node main.js --install
172 verbose exit 6
173 timing npm Completed in 15607ms
174 verbose unfinished npm timer reify 1700950981533
175 verbose unfinished npm timer reify:build 1700950991786
176 verbose unfinished npm timer build 1700950991789
177 verbose unfinished npm timer build:deps 1700950991789
178 verbose unfinished npm timer build:run:install 1700950991797
179 verbose unfinished npm timer build:run:install:node_modules/iobroker.vis 1700950991798
180 verbose code 6
181 error A complete log of this run can be found in: /home/iobroker/.npm/_logs/2023-11-25T22_23_00_740Z-debug-0.log
pi@pi4:~ $
```

**Versions:**  
 - Adapter version: 1.5.1 / 1.4.16
 - JS-Controller version: 5.0.12
 - Node version: 18.18.2
 - Operating system: raspian

Problem is reproduceable

**Comment by mcm1957** [comment]
-- removed as irrelvant for problem --

(was a protectsion problem at windows)

**Comment by Apollon77** [comment]
@mcm1957 This is something else/different ...

**Comment by mcm1957** [comment]
see PR https://github.com/ioBroker/ioBroker.vis/pull/806 which seems to fix this issue.

**Comment by mcm1957** [comment]
fixed with 1.5.2 and later
